### DISCUSSION TITLE
Kyuubi feature stability

### DISCUSSION THREAD

**Discussion by mmuru** [apache/kyuubi#3432] [General]
I am aware of Kyuubi project is still in incubating and noticed that we keep adding the new features. These new features aren't fully functional or incomplete state. This makes Kyuubi is not stable and unable to test end-to-end. I would like to see Kyuubi has set of core features that are fully implemented and fully functional for one engine e.g spark before adding new features. Also, I would like to see Kyuubi roadmap and release details.

**Comment by yaooqinn** [comment]
Thanks, @mmuru for your advice.

> I am aware of Kyuubi project is still in incubating and noticed that we keep adding the new features. 

Thanks to the community and all contributors and users, we are closing to the graduation from incubator now. But, yes, new features are still being actively developed. Maintaining backward compatibility is worth but also hard work.


> Also, I would like to see Kyuubi roadmap and release details.

Do you subscribe our dev mailing list, we discuss and make decision about major changes there. The release process happens there too. And during the release candidate voting process, we ask our users to test on their environment, if any breaking changes happen you can vote -1 to fail the release.
